### PR TITLE
adds CONTRIBUTING document to the main Swift repository 

### PR DESCRIPTION
**Pull request by heckj**
Assisting the Swift OSS contributor experience team. Most of the content is migrated/replicated from https://www.swift.org/contributing/. I also pulled in some of the content from README.md to reference the huge detail for getting started with your first contribution and making an initial pull request.

I intentionally restructured the content to a sentence-per-line (as it's a bit easier to review, comment, and update in Pull Requests). Some latin-isms (.e.g.) were removed and replaced with plain English, and a few sentences converted from future-hopeful-tense to present-tense and active voice.

cc @ahoppen @parispittman

There's additional detail about toolchains, referenced from in here, in the README - that would likely be a good candidate for a separate article and on-par with the "getting started", talking to creating a toolchain from this and related repositories and using them from Xcode.

### CODE REVIEW

**Comment by parispittman** [comment]
@heckj , I think this looks great! Once we have this reviewed and merged in, we'll remove the text from the main guide and start the next part of the journey. :)

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:55]: I don’t think that a new contributor would want to build an entire toolchain. Doing so is its own can of worms and it’s rarely required. So I’d just remove the bullet. (You should be able to inline the Getting started guide link again as well.

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:57]:
```suggestion
We also host answers to [Frequently Asked Questions](/docs/HowToGuides/FAQ.md) that may be of interest.
```

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:92]:
```suggestion
If you are interested in making a large change and feel unsure about its overall effect, please make sure to first discuss the change and reach a consensus through the https://forums.swift.org.
```

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:117]: I have no idea what this sentence is supposed to refer to – we don’t have a commits mailing list and if we do there is no significant traffic on it. Let’s delete it.

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:129]: I don’t know how ancient this paragraph it but most of it isn’t relevant with git and GitHub. I think the only relevant part is

```md
Don’t commit changes authored by others unless they have submitted the change to the project or you have been authorized to submit on their behalf — for example, you work together and your company authorized you to contribute the changes.
```

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:135]: There is no *Community Overview* section in this document.

```suggestion
The license and copyright protections for Swift.org code are called out at the top of every source code file.
```

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:136]: I don’t think it’s that rare to contribute a new source file…

```suggestion
If you contribute a change that includes a new source file, ensure that the header is filled out appropriately.
```

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:186]:
```suggestion
- Code reviews are conducted on GitHub.
```

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:189]:
```suggestion
Code review can be an iterative process, which continues until the change is ready to be merged.
```

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:221]:
```suggestion
1. Code must compile without errors on all supported platforms and should be free of warnings on at least one platform.
```

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:239]: I don’t think this paragraph contributes to anything. CI testing is required for a PR to be merged and I don’t think it’s reasonable for anyone to *watch the CI infrastrucute* throughout the next day. Also, the CI infrastructure doesn’t send failure emails. Let’s just remove it.

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:279]: CI testing prevents you from breaking the build. Let’s remove this.

File: CONTRIBUTING.md
Comment (ahoppen) [inline CONTRIBUTING.md:313]:
```suggestion
Contributions to [llvm-project clone](https://github.com/swiftlang/llvm-project) are governed by the [LLVM Developer Policy](http://llvm.org/docs/DeveloperPolicy.html) and should follow the appropriate [licensing](http://llvm.org/docs/DeveloperPolicy.html#copyright-license-and-patents) and [coding standards](http://llvm.org/docs/CodingStandards.html).
```

**Review by ahoppen — COMMENTED** [review]
Thank you @heckj. Some sections are a little rusty and out-dated but it’s a great start. I added review comments for some of the ancient practices that no longer exist.

File: CONTRIBUTING.md
Comment (heckj) [reply CONTRIBUTING.md:33]: It would probably be better as a github issue template as well

Comment (ahoppen) [reply CONTRIBUTING.md:33]: That’s what I meant: We have GitHub issue forms that ask for all this information individually.

**Comment by ahoppen** [comment]
@heckj I think there are still a few open comments of mine. Can you let me know when you’re addressed them?

File: CONTRIBUTING.md
Comment (heckj) [reply CONTRIBUTING.md]: Should be gone now with the latest commit

File: CONTRIBUTING.md
Comment (heckj) [reply CONTRIBUTING.md]: Should be gone now with the latest commit

**Comment by heckj** [comment]
@ahoppen Ah, sorry - I added the last commit to remove the two additional sections you called out, but didn't resolve those conversations. Wasn't entirely sure of the etiquette there and if I should, or if that would be rude/inappropriate. In either case, the last commit update should have all your comments/suggestions addressed.

**Comment by ahoppen** [comment]
I think you haven’t addressed https://github.com/swiftlang/swift/pull/75654#discussion_r1706174061 yet, for example.

**Comment by heckj** [comment]
I removed the line "Such commits must not break the build." but left the rest in place. Did I misunderstand your suggestion - were you suggesting that the whole second bullet there?
> - You may commit an obvious change without first getting approval.

**Comment by ahoppen** [comment]
Sorry, I think I copied the wrong link. I updated my previous comments to that all that I think haven’t been addressed yet are marked as “Unresolved”.

**Comment by heckj** [comment]
I think that's sorted - thank you for all the suggestions in the diff format for tweaks and updates!

**Comment by ahoppen** [comment]
I think you missed a couple comments that were collapsed behind the “9 hidden conversations”, eg. https://github.com/swiftlang/swift/pull/75654#discussion_r1706167329

File: CONTRIBUTING.md
Comment (heckj) [reply CONTRIBUTING.md:42]: removed bullet and inlined the link to the getting started docs

File: CONTRIBUTING.md
Comment (heckj) [reply CONTRIBUTING.md]: make this a link, not just a raw URL

File: CONTRIBUTING.md
Comment (heckj) [reply CONTRIBUTING.md:104]: dropping both sentences

File: CONTRIBUTING.md
Comment (heckj) [reply CONTRIBUTING.md:116]: removed the non-github pieces and left in the "don't submit the work of others" bit with the caveat

**Comment by heckj** [comment]
Yep, definitely did. Sorry for the back and forth on this, there were quite a few I missed in there.

**Review by ahoppen — APPROVED** [review]
Thank you!

**Comment by ahoppen** [comment]
@swift-ci Please smoke test

**Comment by parispittman** [comment]
Thanks @heckj - and now we are on to the next step of removing this content from the main swift.org/contributing guide. onward! :)

**Pull request merged**
